staging/src/k8s.io/apiserver/pkg/authorization/authorizer/interfaces.go

```diff
@@ -76,6 +77,13 @@ func (f AuthorizerFunc) Authorize(a Attributes) (bool, string, error) {
 	return f(a)
 }
 
+// AuthorizationRulesGetter provides a function that get the list of rules that apply to a given user in a given namespace and error.
+// If an error is returned, the slice of rules may not be complete, but it contains all retrievable rules.  This is done because policy
+// rules are purely additive and policy determinations can be made on the basis of those rules that are found.
+type AuthorizationRulesGetter interface {
+	RulesFor(user user.Info, namespace string) ([]authorizationapi.ResourceRule, []authorizationapi.NonResourceRule, error)
```

Comment:
> Need to articulate how this method works for cluster wide (non-namespaced) rule evaluation. Particularly since this is only being used to fill out a non-namespaced resource.

comments added